analyses/cell-type-dsrct/exploratory_analysis/01-marker-gene-tumor-classification.Rmd

```diff
@@ -0,0 +1,440 @@
+
+```{r}
+for(i in 1:length(umap_df_list_excluded)) {
+  p <- ggplot(umap_df_list_excluded[[i]],
+              aes(x = gene_expression, fill = gene_symbol)) +
+    geom_density() +
+    facet_wrap(vars(gene_symbol), scales = 'free_y') +
+    theme(legend.position = "none") + 
+    ggtitle(names(umap_df_list_excluded)[i])
+  print(p)
+  
+  #save plots
+  ggsave(
+    filename = paste0(
+      '../plots/marker_distribution_',
+      names(umap_df_list)[i],
+      '.png'
+    ),
+    plot = p,
+    width = 6,
+    height = 6,
+    units = 'in',
+    dpi = 150
+  )
+}
+```
+
+Although we see some slight semblance of a bimodal distribution for most marker genes, it is hard to see and we cannot directly compare gene expression values across genes. 
+This would make it hard to identify a cut off to categorize cells as expression or not expressing the marker gene. 
+
+Now we will transform each of the gene expression vectors by generating z-scores. 
+Then we might be able to find a cut off we can use across samples for if marker genes are present in a cell or not. 
+
+```{r}
+umap_df_list_excluded_scaled <- lapply(umap_df_list_excluded, function(x){
+  x   |>
+  dplyr::group_by(gene_symbol) |>
+  # get z-scores for each gene
+  dplyr::mutate(transformed_gene_expression = scale(gene_expression)[, 1]) |>
+  dplyr::ungroup()
+})
+
+```
+
+
+Now we can create the same density plot but looking at z-scores. Interestingly, in some samples, like SPCS000489, we see that ST6GALNAC5 has negative values. Presumably, these are tumor cells but there may be a skew due to the number of tumor cells compared to normal. 
+
+```{r}
+
+for(i in 1:length(umap_df_list_excluded_scaled)) {
+  p <- ggplot(umap_df_list_excluded_scaled[[i]],
+              aes(x = transformed_gene_expression, fill = gene_symbol)) +
+    geom_density() +
+    facet_wrap(vars(gene_symbol), scales = 'free_y') +
+    theme(legend.position = "none") + 
+    ggtitle(names(umap_df_list_excluded_scaled)[i])
+  
+  print(p)
+  
+  #save plots
+  ggsave(
+    filename = paste0(
+      '../plots/marker_distribution_zscaled_',
+      names(umap_df_list)[i],
+      '.png'
+    ),
+    plot = p,
+    width = 6,
+    height = 6,
+    units = 'in',
+    dpi = 150
+  )
+  
+}
+```
+
+It looks like some marker genes have distinct groups of cells with z-score > 0, while other marker genes may not be as informative. 
+
+### Classify tumor cells using marker genes only
+
+Let's try and use the marker gene expression to classify tumor cells. 
+It looks like we could use a cutoff of z-score > 0 to count that cell as a tumor cell. 
+
+We could either count any cell that expresses at least one marker gene > 0 as a tumor cell, or look at the combined expression. 
+Let's start with classifying tumor cells as tumor if any marker gene is present (z-score > 0). 
+This also means the sum of all z-scores > 0. 
```

Comment:
I am confused by this statement. Why would having one marker gene > 0 mean the sum of all is > 0?

Again, I'd probably not want to use the z scores for this, but maybe instead scaling just by range (maybe the 90% quartile or something like that?), but leaving zero as the lower bound? The cutoff may still be hard to draw, but z scoring often breaks when there are not two relatively equal classes. (I should say, one of my current projects is to find a robust way to automatically segment these kinds of distributions.)